src/content/components/FlameChartViewport.js

```diff
@@ -0,0 +1,219 @@
+ *                |___________________________|
+ *        ^       ^                           ^                                ^
+ *        0.0     0.1                         0.6                              1.0
+ *
+ *
+ * viewportLeft = 0.1
+ * viewportRight = 0.6
+ * viewportTop = 30 (in pixels)
+ * viewportLength = viewportRight - viewportLeft
+ * screenWidth = 1000
+ * unitPixel = viewportLength / screenWidth
+ * viewportRight += mouseMoveDelta * unitPixel
+ * viewportLeft += mouseMoveDelta * unitPixel
+ **/
+class FlameChartViewport extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this._mouseWheelListener = this._mouseWheelListener.bind(this);
+    this._mouseDownListener = this._mouseDownListener.bind(this);
+    this._mouseMoveListener = this._mouseMoveListener.bind(this);
+    this._mouseUpListener = this._mouseUpListener.bind(this);
+
+    this._setSize = this._setSize.bind(this);
+
+    /**
+     * TODO - Evaluate whether this state should stay in the component, or go out to
+     * the redux stores. This state information potentially gets changed very frequently
+     * with mouse events.
+     */
+    this.state = {
+      containerWidth: 0,
+      containerHeight: 0,
+      // Unit interval of the profile range.
+      viewportLeft: 0,
+      viewportRight: 1,
+      // Positioning in pixels.
+      viewportTop: 0,
+      viewportBottom: 0,
+    };
+  }
+
+  _setSize() {
+    const rect = this.refs.container.getBoundingClientRect();
+    if (this.state.width !== rect.width || this.state.height !== rect.height) {
+      this.setState({
+        containerWidth: rect.width,
+        containerHeight: rect.height,
+        containerLeft: rect.left,
+        viewportBottom: this.state.viewportTop + rect.height,
+      });
+    }
+  }
+
+  _mouseWheelListener(event) {
+    event.preventDefault();
+    const { containerLeft, containerWidth } = this.state;
+    const mouseCenter = (event.screenX - containerLeft) / containerWidth;
+    const deltaY = event.deltaMode === LINE_SCROLL_MODE
+      ? event.deltaY * SCROLL_LINE_SIZE
+      : event.deltaY;
+
+    const { viewportLeft, viewportRight } = this.state;
+    const viewportLength = viewportRight - viewportLeft;
+    const scale = viewportLength - viewportLength / (1 + deltaY * 0.001);
+    const newBoundsLeft = clamp(0, 1, viewportLeft - scale * mouseCenter);
+    const newBoundsRight = clamp(0, 1, viewportRight + scale * (1 - mouseCenter));
+
+    if (newBoundsLeft === 0 && newBoundsRight === 1) {
+      if (viewportLeft === 0 && viewportRight === 1) {
+        // Do not update if at the maximum bounds.
+        return;
+      }
+    }
+    this.setState({
+      viewportLeft: newBoundsLeft,
+      viewportRight: newBoundsRight,
+    });
+  }
+
+  _mouseDownListener(event) {
+    this.setState({
+      dragX: event.clientX,
+      dragY: event.clientY,
+      isDragging: true,
+    });
+  }
+
+  _mouseMoveListener(event) {
+    if (this.state.isDragging) {
```

Comment:
Hmm... I wonder if `<Draggable>` might get into the way when I end up doing more hit detection type of work for doing tooltips and the side panel? I think I'd like to defer this decision until I get more of that code written, and see if it still makes sense. If it fits in easily with the rest of the code, I think it would be a good decision.